### ISSUE TITLE
 libgobject-2.0.so: cannot open

### ISSUE THREAD

**Issue by oak-tree** [jcupitt/pyvips#31]
Installing on ubnutu 16 

```
 sudo apt install libvips-tools
 sudo apt install libvips
 pip install pyvips
 python
 import pyvips
```

 

>    raise OSError(msg)
> OSError: cannot load library libgobject-2.0.so: libgobject-2.0.so: cannot open shared object file: No such file or directory.  Additionally, ctypes.util.find_library() did not manage to locate a library called 'libgobject-2.0.so'
> >>>

any ideas?

related
https://github.com/stechstudio/libvips-lambda/issues/1

**Comment by jcupitt** [comment]
I think that's probably describing a different issue.

I guess your libgobject is called something else. I see:

```
$ ldd $(which vips) | grep libgob
	libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f0797480000)
$ ls -l /usr/lib/x86_64-linux-gnu/libgob*
-rw-r--r-- 1 root root 681890 Oct  4 14:49 /usr/lib/x86_64-linux-gnu/libgobject-2.0.a
lrwxrwxrwx 1 root root     26 Oct  4 14:49 /usr/lib/x86_64-linux-gnu/libgobject-2.0.so -> libgobject-2.0.so.0.5400.1
lrwxrwxrwx 1 root root     26 Oct  4 14:49 /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 -> libgobject-2.0.so.0.5400.1
-rw-r--r-- 1 root root 342072 Oct  4 14:49 /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0.5400.1
```

On 17.10. What do you get?

**Comment by jcupitt** [comment]
I think I might have found it: the `libgobject-2.0.so` symlink is installed by the -dev package for glib. 

I've changed pyvips to include a version number in the soname -- would you be able to test it? Just install git master. 

Another fix would be to install the `-dev` packages:

```
$ sudo apt install libvips-dev
```

which might also work.

**Comment by oak-tree** [comment]
```
ldd $(which vips) | grep libgob
libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007fa6a0f6f000)
```

installing from master branch works!

Thanks for the quickreply/fix

**Comment by jcupitt** [comment]
I'll push out a new pyvips.

**Comment by jcupitt** [comment]
It's been fixed in 2.2.1, you probably have some other issue. Can you make a dockerfile that reproduces your problem?
